app/src/full/java/io/homeassistant/companion/android/sensors/HealthConnectSensorManager.kt

```diff
@@ -104,13 +105,16 @@ class HealthConnectSensorManager : SensorManager {
                 )
             )
         }
-        totalCaloriesBurnedRequest[TotalCaloriesBurnedRecord.ENERGY_TOTAL]?.let {
+        totalCaloriesBurnedRequest[TotalCaloriesBurnedRecord.ENERGY_TOTAL]?.let { energy ->
             onSensorUpdated(
                 context,
                 totalCaloriesBurned,
-                BigDecimal(it.inKilocalories).setScale(2, RoundingMode.HALF_EVEN),
+                BigDecimal(energy.inKilocalories).setScale(2, RoundingMode.HALF_EVEN),
                 totalCaloriesBurned.statelessIcon,
-                attributes = mapOf("endTime" to LocalDateTime.of(LocalDate.now(), LocalTime.now()).toInstant(ZoneOffset.UTC))
+                attributes = mapOf(
+                    "endTime" to LocalDateTime.of(LocalDate.now(), LocalTime.now()).toInstant(ZoneOffset.UTC),
+                    "package" to totalCaloriesBurnedRequest.dataOrigins.map { it.packageName }
```

Comment:
This returns a `Set` so you're creating a list here, is that intentional? If so maybe it is worth adding a `s` to the attribute name to differentiate it.

Reply:
I made it into a list as I was unsure a `Set` would translate properly in teh attribute so wanted to use something I knew worked :) and I agree adding a `s` here makes sense.